### ISSUE TITLE
Does `--alignSJDBoverhangMin` work for very short reads?

### ISSUE THREAD

**Issue by y9c** [alexdobin/STAR#1807]
I test one example read (`CTGGCTGGAGCAGCGGGACCGCTCCATCCGAGAGAAGCAGAGTGATGATGAGGTTTATGCGCCAGGT`), and run STAR with `--alignSJDBoverhangMin 1` argument. 

The last 3rd base *..TATGCGCCA***G** is the end of exon, and match the `5' ss` of the splicing site. Although the last 2 bases can match into the intro, it might be more reasonable to open a splicing event for these bases. 

The output from STAR:

```
CTGGCTGGAGCAGCGGGACCGCTCCATCCGAGAGAAGCAGAGTGATGATGAGGTTTATGCGCCAGGT
|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
CTGGCTGGAGCAGCGGGACCGCTCCATCCGAGAGAAGCAGAGTGATGATGAGGTTTATGCGCCAGGT
```

Expected output:

```
CTGGCTGGAGCAGCGGGACCGCTCCATCCGAGAGAAGCAGAGTGATGATGAGGTTTATGCGCCAG........GT
|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||        ||
CTGGCTGGAGCAGCGGGACCGCTCCATCCGAGAGAAGCAGAGTGATGATGAGGTTTATGCGCCAG........GT
```

*... is the intron*

**Comment by alexdobin** [comment]
```--alignSJDBoverhangMin 1``` only affects annotated junctions (i.e. those provided in the GTF).
For unannotated junctions, it's practically impossible to find splices with very short overhangs.

**Comment by y9c** [comment]
Yes. This junction is annotated in the gtf file and I build the index with this annotation file.

**Comment by alexdobin** [comment]
This is not supposed to happen, the spliced alignment should be preferred over unspliced for annotated junctions.

**Comment by y9c** [comment]
Thank you. Is this a bug? I tested it on GRCm39 genome.

**Comment by alexdobin** [comment]
It's unlikely this is a bug.
Could you please STAR SAM lines for this read?
